app/Http/Controllers/ProfileController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\User;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Session;
+
+class ProfileController extends Controller
+{
+    public function profileView($username)
```

Comment:
Since you have a user_profile db table setup. You should set up an [eloquent](https://laravel.com/docs/5.2/eloquent) relationship between user and user_profile. This will allow you to just pass a user object and you can access user_profile from it. No need to send username